### ISSUE TITLE
How do you submit an empty field when `wrap` is a double quote?

### ISSUE THREAD

**Issue by tandrewnichols** [mrodrig/json-2-csv#115]
## Background Information
- Module Version: `3.5.0`
- Node/Browser Version: `9.11.2`

The issue I'm reporting is with:
- [ ] json2csv
- [x] csv2json

I have...
- [x] searched to see if an issue has already been reported.
- [x] verified that my JSON/CSV data is valid (using something like http://jsonlint.com or https://csvlint.io/).
- [x] tried upgrading to the latest version of json-2-csv (since the issue may already be fixed).

## Expected/Actual Behavior
We're getting CSV data back from an third party where each "column" value is wrapped in double quotes. For example: `"foo","bar","baz"`. When we get an empty string in one of those fields, it gets parsed as part of the next column. So...

```
"a","b","c"
"foo","","bar"
```

gets parsed as

```
[{ a: "foo", b: "\",\"bar",c: undefined }]
```

which can throw off the alignment of later rows. I see from some research that the way to escape a double quote in a csv is a second double quote, i.e. `""` and that seems to be what you're going for [here](https://github.com/mrodrig/json-2-csv/blob/master/src/csv2json.js#L169), but shouldn't that only be _within_ another quote? Like `"This is a ""double quote"""`? Like I said, this data is coming from a third party (salesforce), so I can't alter it. Is there some option that handles this case or is this a case that _should_ work and isn't? How can I make empty values (`""`) actually be empty without doing some sort of gross `.replace(/,"",/g, ',,')`?

[Runkit](https://npm.runkit.com/json-2-csv) example:

```
var json2Csv = require("json-2-csv")
let csv = `"a","b","c"
"foo","","bar"
"baz","qux","hello"`
json2Csv.csv2jsonAsync(csv).then((data) => console.log(data))
```

**Comment by tandrewnichols** [comment]
Incidentally, it _does_ work when the empty value is the _first_ in a line, like:

```
let csv = `"a","b","c"
"","foo","bar"
```

**Comment by mrodrig** [comment]
Hi @tandrewnichols. Thanks for reporting that! After taking a look, it's definitely a bug. I was able to figure out exactly what was going wrong and already have a fix. Once the CI tests finish running, I'll merge the fix into master and will push out a new version on NPM with that bug fixed. It'll be out as `3.5.1` on NPM.

Thanks again for reporting this and sorry for any inconvenience this may have caused!

**Comment by tandrewnichols** [comment]
Thanks, @mrodrig! Very fast turn around. I appreciate the time and work.

**Comment by tandrewnichols** [comment]
@mrodrig I don't know if this is a regression because of the above fix, but I'm also (or now) seeing a similar issue with double quotes in the last column of a row. For example:

```
"a","b","c"
"foo","bar",""
"baz","quux","blah"
```

Parses as:

```
[{
    a: 'foo'
    b: 'bar'
    c: 'baz'
}]
```

**Comment by mrodrig** [comment]
@tandrewnichols Hmm, I was pretty sure that I had tested that case, but maybe something got messed up when I was shuffling changes around to different change lists in my editor. I should be able to take a look at this tomorrow. I don't think it will be a difficult fix, so hopefully I'll be able to get a new release out on NPM.  Sorry about that. I'll keep you updated with any progress that I make.  Thanks!

**Comment by mrodrig** [comment]
@tandrewnichols I just published `3.5.3` on NPM which fixes that issue. Thanks again for letting me know!

**Comment by tandrewnichols** [comment]
👍
